### PR TITLE
feat(shared-link-settings-modal): add hideVanityNameSection prop

### PR DESCRIPTION
**Pull request by nickgriffin1**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @nickgriffin1 has signed the CLA. Thanks for the pull request!

File: src/features/shared-link-settings-modal/__tests__/SharedLinkSettingsModal-test.js [hunk 1]
```diff
@@ -24,9 +24,6 @@ describe('features/shared-link-settings-modal/SharedLinkSettingsModal', () => {
     const isDirectLinkAvailable = true;
     const isDirectLinkUnavailableDueToDownloadSettings = true;
```
Comment (nickgriffin1) [inline src/features/shared-link-settings-modal/__tests__/SharedLinkSettingsModal-test.js]: Fixed this file

File: src/features/shared-link-settings-modal/SharedLinkSettingsModal.js [hunk 2]
```diff
@@ -220,9 +221,19 @@ class SharedLinkSettingsModal extends Component {
...
+        const {
+            canChangeVanityName,
+            hideVanityNameSection,
+            serverURL,
+            vanityNameInputProps,
+            warnOnPublic = false,
+        } = this.props;
         const { vanityNameError, isVanityEnabled } = this.state;
 
+        if (hideVanityNameSection) {
```
Comment (ChenCodes) [inline src/features/shared-link-settings-modal/SharedLinkSettingsModal.js]: Nitpick: Could we do this instead? ```{!hideVanityNameSection ? <VanityName... /> : null}```

Comment (nickgriffin1) [reply src/features/shared-link-settings-modal/SharedLinkSettingsModal.js]: That's exactly how I first wrote it then I changed it to this to be more consistent with other box react `render()` methods.

**Pull request merged**